### PR TITLE
Add lint make target

### PR DESCRIPTION
**Pull request by cartermckinnon**
**Description of changes:**

Adds a `make` target called `lint` which will be added to the CI, failing if the source files are not of the proper format or if they have bug-checking failures.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Testing Done**

This will fail because I haven't opened a PR standardizing the format yet :) 
```
make lint
```

### CODE REVIEW

File: Makefile
Comment (cartermckinnon) [inline Makefile:52]: There's not a simple way to do shebang matching  for shellcheck, but shfmt will catch some syntax issues anyway. probably good enough for the time being.

File: Makefile
Comment (cartermckinnon) [inline Makefile:53]: I started with `error` severity, because there's only one of those right now; while there are 18 warnings. I can raise the bar once we address more of them (or decide whether a lower severity bar makes sense).

File: Makefile
Comment (cartermckinnon) [inline Makefile:52]: There's not a simple way to do shebang matching for shellcheck, but shfmt will catch some syntax issues anyway. probably good enough for the time being.

**Comment by cartermckinnon** [comment]
I went ahead and added `make lint` to the CI workflow, so that check will need to be overridden to merge this.

I'll open two subsequent PR's:
1. Apply formatting.
2. Fix the 1 error-level `shellcheck` finding.

I don't want to cause headache for the PR's that are close to merge, so I'll wait until #938 and #1066 are merged to move forward.

**Review by mmerkes — APPROVED** [review]

**Pull request merged**
